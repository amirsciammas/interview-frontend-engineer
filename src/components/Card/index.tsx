import React from 'react';
import * as classes from './card.module.css';

interface Props {
  children: React.ReactChild
  additionalStyles?: Record<string, any>;
}

function Card(props: Props) {
  const { children, additionalStyles } = props;
  return (
    <div className={classes.card} style={additionalStyles}>
      {children}
    </div>
  );
}

Card.defaultProps = {
  additionalStyles: {},
};

export default Card;
