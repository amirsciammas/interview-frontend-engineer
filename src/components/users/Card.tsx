import { useNavigate } from 'react-router-dom';
import './card.css';

type Props = {
    data: any;
}

const getRandomColor = () => {
    const randomColor: string = Math.floor(Math.random()*16777215).toString(16);
    return randomColor === '000000' || randomColor === 'ffffff' ? '00DECA' : randomColor;
}

const Card = ({ data }: Props) => {
    const navigate = useNavigate();

    const handleSeePosts = (id: number, name: string) => {
        navigate(`/${name}/post/${id}`);
    }

    const getAndSetInitials = (name: string) => {
        let initials = '';
        const names = name.split(" ");
        if (names.length === 1) {
            initials = names[0].substring(0, 1).toUpperCase();
        } else if (names.length > 1) {
            names.forEach((_, i) => {
                initials += names[i].substring(0, 1).toUpperCase();
            });
        }
        return initials;
    }

    return (
        <div className='card-container'>
            <div className='card-avatar' style={{ background: '#' + getRandomColor() }}>
                <div className='card-name'>{getAndSetInitials(data?.name)}</div>
            </div>
            <div className='card-detail'>
                <div className='card-title'>{data?.name}</div>
                <div className='card-sub-title'>{data?.email}</div>
                <div className='card-link' onClick={() => handleSeePosts(data?.id, data?.name)}>See posts</div>
            </div>
        </div>
    )
}

export default Card;