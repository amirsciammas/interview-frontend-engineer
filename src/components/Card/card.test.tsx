import React from 'react';
import renderer from 'react-test-renderer';
import Card from './index';

test('card component renders correctly', () => {
  const component = renderer.create(<Card>Hello!</Card>);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
