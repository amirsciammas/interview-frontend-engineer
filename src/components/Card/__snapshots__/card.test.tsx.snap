// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card component renders correctly 1`] = `
<div
  style={Object {}}
>
  Hello!
</div>
`;
